--- cmake-2.4.2/Modules/FindRuby.cmake.fedora	2006-07-11 10:20:14.000000000 -0600
+++ cmake-2.4.2/Modules/FindRuby.cmake	2006-07-11 10:20:38.000000000 -0600
@@ -9,6 +9,7 @@
 
 SET(RUBY_POSSIBLE_INCLUDE_PATHS
-  /usr/lib/ruby/1.8/i386-linux
+  /usr/lib/ruby/*/i386-linux
+  /usr/lib64/ruby/*/x86_64-linux
   )
 
 SET(RUBY_POSSIBLE_LIB_PATHS
@@ -19,12 +20,12 @@
   ${RUBY_POSSIBLE_INCLUDE_PATHS})
 
 FIND_LIBRARY(RUBY_LIBRARY
-  NAMES ruby1.8
+  NAMES ruby
   PATHS ${RUBY_POSSIBLE_LIB_PATHS}
   )
 
 FIND_PROGRAM(RUBY_EXECUTABLE
-  NAMES ruby1.8
+  NAMES ruby
   PATHS
   /usr/bin
   /usr/local/bin
--- cmake-2.4.2/Modules/FindSWIG.cmake.fedora	2006-07-11 10:21:11.000000000 -0600
+++ cmake-2.4.2/Modules/FindSWIG.cmake	2006-07-11 10:22:47.000000000 -0600
@@ -7,13 +7,11 @@
 SET(SWIG_FOUND FOOBAR)
 FIND_PATH(SWIG_DIR
   SWIGConfig.cmake
-  /usr/share/swig1.3
-  /usr/lib/swig1.3
+  /usr/share/swig/*
   /usr/local/share/swig1.3)
 FIND_PATH(SWIG_DIR
   swig.swg
-  /usr/share/swig1.3
-  /usr/lib/swig1.3
+  /usr/share/swig/*
   /usr/local/share/swig1.3)
 IF(EXISTS ${SWIG_DIR})
   IF("x${SWIG_DIR}x" STREQUAL "x${CMAKE_ROOT}/Modulesx")
