diff -up cmake-2.6.1/Source/cmELF.cxx.parens cmake-2.6.1/Source/cmELF.cxx
--- cmake-2.6.1/Source/cmELF.cxx.parens	2008-08-01 10:34:52.000000000 -0500
+++ cmake-2.6.1/Source/cmELF.cxx	2008-08-26 14:58:23.000000000 -0500
@@ -884,8 +884,8 @@ cmELF::StringEntry const* cmELF::GetSONa
 cmELF::StringEntry const* cmELF::GetRPath()
 {
   if(this->Valid() &&
-     this->Internal->GetFileType() == cmELF::FileTypeExecutable ||
-     this->Internal->GetFileType() == cmELF::FileTypeSharedLibrary)
+     (this->Internal->GetFileType() == cmELF::FileTypeExecutable ||
+     this->Internal->GetFileType() == cmELF::FileTypeSharedLibrary) )
     {
     return this->Internal->GetRPath();
     }
@@ -899,8 +899,8 @@ cmELF::StringEntry const* cmELF::GetRPat
 cmELF::StringEntry const* cmELF::GetRunPath()
 {
   if(this->Valid() &&
-     this->Internal->GetFileType() == cmELF::FileTypeExecutable ||
-     this->Internal->GetFileType() == cmELF::FileTypeSharedLibrary)
+     (this->Internal->GetFileType() == cmELF::FileTypeExecutable ||
+     this->Internal->GetFileType() == cmELF::FileTypeSharedLibrary) )
     {
     return this->Internal->GetRunPath();
     }
