Index: Tests/CMakeLists.txt
===================================================================
RCS file: /cvsroot/CMake/CMake/Tests/CMakeLists.txt,v
retrieving revision 1.148
retrieving revision 1.149
diff -u -r1.148 -r1.149
--- Tests/CMakeLists.txt	29 Oct 2009 16:45:59 -0000	1.148
+++ Tests/CMakeLists.txt	30 Oct 2009 16:08:10 -0000	1.149
@@ -1251,8 +1251,8 @@
 
   set(regex "(Problems when submitting via S*CP")
   set(regex "${regex}|Error message was: ")
-  set(regex "${regex}(Couldn.t resolve host ")
-  set(regex "${regex}|couldn.t connect to host")
+  set(regex "${regex}([Cc]ould *n.t resolve host")
+  set(regex "${regex}|[Cc]ould *n.t connect to host")
   set(regex "${regex}|libcurl was built with SSL disabled. https: not supported)")
   set(regex "${regex}|Submission method .xmlrpc. not compiled into CTest")
   set(regex "${regex}|Submission successful)")
