--- cmake-3.10.1/Utilities/KWIML/include/kwiml/abi.h.old	2017-12-30 06:41:48.775310791 +0000
+++ cmake-3.10.1/Utilities/KWIML/include/kwiml/abi.h	2017-12-30 06:41:56.332310127 +0000
@@ -468,7 +468,7 @@
 # define KWIML_ABI_ENDIAN_ID KWIML_ABI_ENDIAN_ID_LITTLE
 
 /* RISC-V */
-#elif defined(__riscv__)
+#elif defined(__riscv)
 # define KWIML_ABI_ENDIAN_ID KWIML_ABI_ENDIAN_ID_LITTLE
 
 /* Unknown CPU */
