--- cmake-2.8.0-rc1/Modules/FindVTK.cmake.vtk64	2009-09-25 12:29:20.000000000 -0600
+++ cmake-2.8.0-rc1/Modules/FindVTK.cmake	2009-09-29 10:33:20.506713352 -0600
@@ -45,10 +45,15 @@
   SET(VTK_DIR_SEARCH "")
   FOREACH(dir ${VTK_DIR_SEARCH2})
     SET(VTK_DIR_SEARCH ${VTK_DIR_SEARCH}
+      ${dir}/../lib64/vtk-5.4
       ${dir}/../lib/vtk-5.4
+      ${dir}/../lib64/vtk-5.2
       ${dir}/../lib/vtk-5.2
+      ${dir}/../lib64/vtk-5.1
       ${dir}/../lib/vtk-5.1
+      ${dir}/../lib64/vtk-5.0
       ${dir}/../lib/vtk-5.0
+      ${dir}/../lib64/vtk
       ${dir}/../lib/vtk
       )
   ENDFOREACH(dir)
