--- cmake-2.8.5-rc2/Modules/FindSWIG.cmake.swig	2011-06-15 09:18:34.000000000 -0600
+++ cmake-2.8.5-rc2/Modules/FindSWIG.cmake	2011-06-20 09:24:51.200421179 -0600
@@ -27,7 +27,7 @@
 # (To distribute this file outside of CMake, substitute the full
 #  License text for the above reference.)
 
-FIND_PROGRAM(SWIG_EXECUTABLE swig2.0 swig)
+FIND_PROGRAM(SWIG_EXECUTABLE NAMES swig2.0 swig)
 
 IF(SWIG_EXECUTABLE)
   EXECUTE_PROCESS(COMMAND ${SWIG_EXECUTABLE} -swiglib
