diff -up cmake-3.7.2/Utilities/KWIML/test/CMakeLists.txt.format cmake-3.7.2/Utilities/KWIML/test/CMakeLists.txt
--- cmake-3.7.2/Utilities/KWIML/test/CMakeLists.txt.format	2017-01-13 07:05:42.000000000 -0700
+++ cmake-3.7.2/Utilities/KWIML/test/CMakeLists.txt	2017-02-27 11:09:30.789844778 -0700
@@ -10,7 +10,7 @@ endif()
 # Suppress printf/scanf format warnings; we test if the sizes match.
 foreach(lang C CXX)
   if(KWIML_LANGUAGE_${lang} AND CMAKE_${lang}_COMPILER_ID STREQUAL "GNU")
-    set(CMAKE_${lang}_FLAGS "${CMAKE_${lang}_FLAGS} -Wno-format")
+          set(CMAKE_${lang}_FLAGS "${CMAKE_${lang}_FLAGS} -Wno-format -Wno-format-security")
   endif()
 endforeach()
 
