Index: Modules/FindJNI.cmake
===================================================================
RCS file: /cvsroot/CMake/CMake/Modules/FindJNI.cmake,v
retrieving revision 1.45
diff -u -r1.45 FindJNI.cmake
--- Modules/FindJNI.cmake	22 Oct 2009 10:57:00 -0000	1.45
+++ Modules/FindJNI.cmake	29 Oct 2009 16:25:07 -0000
@@ -195,14 +195,13 @@
 )
 
 FIND_PATH(JAVA_INCLUDE_PATH2 jni_md.h 
-  ${JAVA_AWT_INCLUDE_DIRECTORIES}
+  ${JAVA_INCLUDE_PATH}
   ${JAVA_INCLUDE_PATH}/win32
   ${JAVA_INCLUDE_PATH}/linux
   ${JAVA_INCLUDE_PATH}/freebsd
 )
 
 FIND_PATH(JAVA_AWT_INCLUDE_PATH jawt.h
-  ${JAVA_AWT_INCLUDE_DIRECTORIES}
   ${JAVA_INCLUDE_PATH}
 )
 
